[material-ui][FilledInput] Remove unapplied classes from filledInputClasses interface and add missing classes to root

sai6855 opened the pull request:
<!-- Thanks so much for your PR, your contribution is appreciated! ❤️ -->

[Documentation says](https://mui.com/material-ui/api/filled-input/#classes) following classes are applied when necessary conditions are met but actually none of the classes are applied. I've written test case to validate same [here](https://github.com/mui/material-ui/pull/42082/files#diff-0c29a4032ec827ace42291c4c833c034250496dce6d3ea013bdc484918be4d33)

1.  .MuiFilledInput-sizeSmall
2.   .MuiFilledInput-multiline
3.   .MuiFilledInput-adornedEnd
4.   .MuiFilledInput-adornedStart
5.   .MuiFilledInput-inputSizeSmall
6.   .MuiFilledInput-inputMultiline
7.   .MuiFilledInput-inputAdornedStart
8.   .MuiFilledInput-inputAdornedEnd
9.   .MuiFilledInput-inputTypeSearch

we can do either of the following

1. to remove above classes from filledInputClasses interface and from docs. that's what this PR does
2. to add above classes to applicable elements and don't remove any classes from filledInputClasses interface

@DiegoAndai which approach you recommend 




- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/mui/material-ui/blob/HEAD/CONTRIBUTING.md#sending-a-pull-request).

DiegoAndai commented:
This is a difficult one. `InputBase` also has these classes, should that component host them, or should each of the inputs (Standard, Outlined, Filled) host them 🤔

With the current architecture, it seems more useful for users that each input hosts their own classes. This might also change when we update the styles and theme structure, so we might want to avoid breaking changes until then.

What do you think @aarongarciah

sai6855 commented:
Hey, I think most of the above classes selector behavior can be achieved by combining the FilledInput root class and the `has` selector. Even if we add them now, they can be deprecated and removed in the future.

aarongarciah commented:
The problem is also present in `Input`, `FilledInput`, and `OutlineInput`. So whatever we land on we should apply changes on them, too.

@DiegoAndai I'm inclined to think that every input should host their own classes so consumers can target them individually. If they want to target elements or states common to all inputs, they can target the `InputBase` classes.

@sai6855 even when `:has` is widely supported, I think is very valuable for consumers to have these static classes they can target as part of the components' public APIs. By removing classes, consumers would need to know about components' internal DOM structure and other details that classes help abstract.

sai6855 commented:
>By removing classes, consumers would need to know about components' implementation details that classes help abstract.


@aarongarciah I'm not sure if you are aware of this issue https://github.com/mui/material-ui/issues/41282 but TL DR; of the issue is, we are deprecating composed classes which can be achieved through css selctors. For example `.MuiStepConnector-lineHorizontal` is deprecated in favour of `.MuiStepConnector-horizontal > .MuiStepConnector-line`, so with this change, consumers of compoents are expected to know internal structure of componets. 

docs link for same: https://deploy-preview-41740--material-ui.netlify.app/material-ui/migration/migrating-from-deprecated-apis/#composed-css-classes-8

aarongarciah commented:
@sai6855 TIL! Thanks for bringing it to my attention. Then removing composed classes that are unused makes sense. Forget my previous comment except:

> The problem is also present in `Input`, `FilledInput`, and `OutlineInput`.

DiegoAndai commented:
> we can do either of the following
>
> 1. to remove above classes from filledInputClasses interface and from docs. that's what this PR does
> 2. to add above classes to applicable elements and don't remove any classes from filledInputClasses interface
>
> @DiegoAndai which approach you recommend

I think option 3:

From that list, remove the composed classes, and add the ones that are not composed. So for example:

- Add  `.MuiFilledInput-sizeSmall`
- Remove `.MuiFilledInput-inputAdornedStart`. No need to deprecate it if it never worked

Does that make sense?

aarongarciah commented:
@DiegoAndai yes, I think that makes sense

sai6855 commented:
@DiegoAndai i went with your suggestion. commit for change -> https://github.com/mui/material-ui/pull/42082/commits/29454775e6fd88c55a9fcf8b697488a8a64f990d

Summary of changes : 

1. Removed following classes

- .MuiFilledInput-inputSizeSmall
- .MuiFilledInput-inputMultiline
- .MuiFilledInput-inputAdornedStart
- .MuiFilledInput-inputAdornedEnd
- .MuiFilledInput-inputTypeSearch

2. Added following classes to `FilledInput` root

- .MuiFilledInput-sizeSmall
- .MuiFilledInput-multiline
- .MuiFilledInput-adornedEnd
- .MuiFilledInput-adornedStart
- .MuiFilledInput-hiddenLabel

sai6855 commented:
Ping @DiegoAndai

DiegoAndai reviewed (approved):
Thanks @sai6855!
